scarlet/component.py

```diff
@@ -123,31 +123,37 @@ def coord(self):
     def sed(self):
         """Numpy view of the component SED
         """
-        return self._sed.data.detach().numpy()
+        return self._sed
 
     @property
     def morph(self):
         """Numpy view of the component morphology
         """
-        return self._morph.data.detach().numpy()
+        return self._morph
 
-    def get_model(self, numpy=True):
+    def get_model(self, sed=None, morph=None):
```

Comment:
When do you need the ability to call this with some other sed and morphology?

Reply:
This is what I was talking about today with the way autograd calculates gradients. You basically need to setup `nabla = grad(loss_function, (a long tuple here))` to calculate the partial derivatives, and call with with `nabla(sed1, sed2, ..., morph1, morph2, ...)`. So inside the loss function you have to build the model with the seds's and morphologies passed to the loss function. See https://github.com/fred3m/scarlet/blob/auto/scarlet/blend.py#L111-L146. 

This is different than pytorch, since it knew which parameters to use for the partial derivatives.

Reply:
I understand that `nabla` requires all parameters, but `blend.get_model` should work identically without arguments. *I think* you only need to declare the parameters for autograd, but you don't need to strictly pass them.

Reply:
In essence, replace https://github.com/fred3m/scarlet/blob/auto/scarlet/blend.py#L136-L140 with
```
model = self.get_model()
```

Reply:
Unfortunately no. Inside the loss function I inserted
```
print(type(seds[0]))
print(type(self.sources[0].sed))
```
at https://github.com/fred3m/scarlet/blob/auto/scarlet/blend.py#L139 and the output was

```
<class 'autograd.numpy.numpy_boxes.ArrayBox'>
<class 'numpy.ndarray'>
```

What happens is the parameters passed to `nabla` are wrapped in the `ArrayBox` class, which is what autograd uses to propagate the gradients. Architecturally this is very different than the `Jax` and `pytorch` implementations.

Reply:
Understood, thanks for checking